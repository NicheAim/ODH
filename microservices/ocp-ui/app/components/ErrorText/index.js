/**
 *
 * ErrorText
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const ErrorText = styled.span`
  color: rgb(244, 67, 54);
  margin-left: ${({ marginLeft }) => marginLeft};
  padding-left: ${({ paddingLeft }) => paddingLeft};
`;

ErrorText.propTypes = {
  marginLeft: PropTypes.string,
  paddingLeft: PropTypes.string,
};
ErrorText.defaultProps = {
  marginLeft: '1vw',
  paddingLeft: '0.5vw',
};

export default ErrorText;
