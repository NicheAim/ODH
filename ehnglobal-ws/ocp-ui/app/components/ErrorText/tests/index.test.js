import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import ErrorText from '../index';

configure({ adapter: new Adapter() });

describe('<ErrorText />', () => {
  describe('snapshot tests', () => {
    it('should have children', () => {
      // Arrange
      const children = (<span>test</span>);

      // Act
      const renderedComponent = shallow(<ErrorText>{children}</ErrorText>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('structural tests', () => {
    it('should have children', () => {
      // Arrange
      const children = (<span>test</span>);

      // Act
      const renderedComponent = shallow(<ErrorText>{children}</ErrorText>);

      // Assert
      expect(renderedComponent.contains(children)).toEqual(true);
    });
  });

  describe('style tests', () => {
    it('should have styles', () => {
      // Arrange
      const children = (<span>test</span>);

      // Act
      const renderedComponent = shallow(<ErrorText>{children}</ErrorText>);

      // Assert
      expect(renderedComponent).toHaveStyleRule('color', 'rgb(244, 67, 54)');
      expect(renderedComponent).toHaveStyleRule('padding-left', '0.5vw');
      expect(renderedComponent).toHaveStyleRule('margin-left', '1vw');
    });
  });
});
