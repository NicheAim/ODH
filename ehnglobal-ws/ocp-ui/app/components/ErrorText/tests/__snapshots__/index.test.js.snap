// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorText /> snapshot tests should have children 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorText__ErrorText
    marginLeft="1vw"
    paddingLeft="0.5vw"
  >
    <span>
      test
    </span>
  </ErrorText__ErrorText>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        test
      </span>,
      "className": "ErrorText__ErrorText-jrrceQ fyYNAz",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "span",
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          test
        </span>,
        "className": "ErrorText__ErrorText-jrrceQ fyYNAz",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
